lib/hdt.js

```diff
@@ -67,6 +67,40 @@ HdtDocumentPrototype.searchTerms = function (options) {
     }, this);
   });
 };
+
+// Returns the header of the HDT document as an array of triples.
+HdtDocumentPrototype.readHeader = function () {
+  if (this.closed) return Promise.reject(new Error('The HDT document cannot be read because it is closed'));
+  return new Promise((resolve, reject) => {
+    this._readHeader((error, triples, totalCount) => {
+      if (error)
+        reject(error);
+      else
+        resolve(triples);
+    }, this);
+  });
+};
```

Comment:
Nit, but the above takes up a lot of space for what it is doing.
Therefore, I usually go for  `this._readHeader(header, e => e ? reject(e) : resolve(), this)`